state/factory/historyfactory.go

```diff
@@ -46,5 +46,6 @@ func (hReader *historyStateReader) States(opts ...protocol.StateOption) (uint64,
 
 // ReadView reads the view
 func (hReader *historyStateReader) ReadView(name string) (uint64, interface{}, error) {
-	return hReader.sf.ReadView(name)
+	_, view, err := hReader.sf.ReadView(name)
+	return hReader.height, view, err
```

Comment:
rebase and revert this, the correct fix has been checked in

Reply:
this pr based on v1.1.1

Reply:
ok, then need to cherry-pick the fix https://github.com/iotexproject/iotex-core/commit/4456bfa417108d0985a8a6e4ef23cc3bc60adc4f